### PR TITLE
Fix style without selector not finding target type

### LINKED ISSUE #18024: Adding a style without selector does not locate the target type.
### Describe the bug

When defining styles, if the user wants to reference the Styles owner in a setter, not providing a selector will reference the owner, in other words, it's implicit "self". Currently this doesn't work and crashes when set in xaml with the error
```Could not determine target type of Setter Line Y, position X```

### To Reproduce

Add a style with no selector to a control.
```
  <UserControl.Styles>
    <Styles>
      <Style>
        <Setter Property="IsVisible" Value="False"/>
      </Style>
    </Styles>
  </UserControl.Styles>
```

### Expected behavior

The setter finds the target type, which is the Styles owner, and sets IsVisible property on the control.
This issue can be worked around by setting `x:SetterTargetType` on the setter, but as there's an explicit test available for this in code form, xaml should also behave the same: https://github.com/AvaloniaUI/Avalonia/blob/master/tests/Avalonia.Base.UnitTests/Styling/StyleTests.cs#L81 .

### Avalonia version

11.2.3

### OS

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by MrJul**
## What does the pull request do?
This PR fixes the XAML compiler not being able to find the target type of a style without selector.
See #18024 for details.
Unit tests have been added.

## Fixed issues
 - Fixes #18024

### CODE REVIEW

File: tests/Avalonia.Markup.Xaml.UnitTests/Xaml/StyleTests.cs [hunk 1]
```diff
@@ -739,5 +737,28 @@ public void Fails_Use_Classes_In_Setter_When_Selector_Is_Complex()
...
+                <Window xmlns="https://github.com/avaloniaui">
+                    <Window.Styles>
+                        {styleStart}
+                            <Setter Property="WindowStartupLocation" Value="CenterScreen" />
+                        {styleEnd}
+                    </Window.Styles>
+                </Window>
+                """);
+
+            Assert.Equal(WindowStartupLocation.CenterScreen, window.WindowStartupLocation);
```
Comment (maxkatz6) [inline tests/Avalonia.Markup.Xaml.UnitTests/Xaml/StyleTests.cs:761]: Use `Tag` or something less dependent on default values being changed?

Comment (MrJul) [reply tests/Avalonia.Markup.Xaml.UnitTests/Xaml/StyleTests.cs]: Good point! I wanted a property from `Window` itself, to prove that we're not falling back to `Control` or equivalent. Changed to `Window.Title`.

File: src/Markup/Avalonia.Markup.Xaml.Loader/CompilerExtensions/Transformers/AvaloniaXamlIlSelectorTransformer.cs [hunk 2]
```diff
@@ -32,8 +32,26 @@ public IXamlAstNode Transform(AstTransformationContext context, IXamlAstNode nod
...
             if (pn == null)
+            {
+                // We already went through this node
+                if (context.ParentNodes().FirstOrDefault() is AvaloniaXamlIlTargetTypeMetadataNode metadataNode
+                    && metadataNode.Value == on)
+                {
+                    return node;
+                }
+
+                if (FindStyleParentObject(context) is { } parentObjectNode)
```
Comment (maxkatz6) [inline src/Markup/Avalonia.Markup.Xaml.Loader/CompilerExtensions/Transformers/AvaloniaXamlIlSelectorTransformer.cs:45]: How would it work, if Style without selector is inside of ControlTheme?

Comment (maxkatz6) [reply src/Markup/Avalonia.Markup.Xaml.Loader/CompilerExtensions/Transformers/AvaloniaXamlIlSelectorTransformer.cs:45]: Of I suppose ControlTheme case would be already handled at this point above. Same with nested styles. Would make sense to add test case for these.

Other than that, I think lookup for StyledObject would make sense here. But can be too restrictive in some corner cases.

Comment (MrJul) [reply src/Markup/Avalonia.Markup.Xaml.Loader/CompilerExtensions/Transformers/AvaloniaXamlIlSelectorTransformer.cs:45]: Good catch.

Styles without a selector in `ControlTheme` weren't failing at compile time (even before this PR), since we're in a `Style` scope with a proper target type thanks to `AvaloniaXamlIlControlThemeTransformer`. They were failing at runtime though.

I've made sure the XAML compiler now handles this case properly, with proper unit tests. I've also restricted the parent to `StyledElement`.

**Review by maxkatz6 — APPROVED** [review]
